.github/workflows/rust-ci.yaml

```diff
@@ -0,0 +1,53 @@
+name: Rust CI
+
+on:
+  merge_group:
+  pull_request:
+  push:
+    # We need to explicitly include tags because otherwise when adding
+    # `branches-ignore` it will only trigger on branches.
+    tags:
+      - '*'
+    branches-ignore:
+      # Ignore pushes to merge queues.
+      # We only want to test the merge commit (`merge_group` event), the hashes
+      # in the push were already tested by the PR checks
+      - 'gh-readonly-queue/**'
+      - 'dependabot/**'
+  workflow_dispatch:
+
+jobs:
+  test:
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Fetch sources
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Run tests
+        uses: frequenz-floss/gh-action-cargo-test@v1.0.0
+
+  release:
+    runs-on: ubuntu-latest
+    if: startsWith(github.ref, 'refs/tags/v')
```

Comment:
maybe check that it is a push too?

Reply:
Why also on push?

Reply:
Why not? :laughing: 

I'm not 100% sure, but I guess for `workflow_run` we want to run it too, but we should create a release for a tag only once, so if it already ran for a push, we might get errors. Not sure about crates but pretty sure pypi will complain.

I guess for PRs and merge queues this should probably be never triggered (technically with PRs I think it could if you are doing a PR from a branch with a tag to another branch, but should be pretty rare, maybe?).

Reply:
makes sense :)